desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -45,40 +50,160 @@ def slack_events(request):
 
-  
-def say_hi_user(channel, user_id):
-  """Bot sends Hi<username> message in a specific channel"""
+    if path == '/hue/gist' and ENABLE_GIST_PREVIEW.get():
+      gist_uuid = queryid_or_uuid.split('=')[1]
+      gist_doc = _get_gist_document(uuid=gist_uuid)
+      gist_doc_data = json.loads(gist_doc.data)
+
+      statement = gist_doc_data['statement_raw']
+      created_by = gist_doc.owner.get_full_name() or gist_doc.owner.username
+      dialect = gist_doc.extra.capitalize()
+      
+      payload = make_gist_payload(item['url'], statement, dialect, created_by)
+      response = slack_client.chat_unfurl(channel=channel_id, ts=message_ts, unfurls=payload)
+      if not response['ok']:
+        raise PopupException(_("Cannot unfurl gist link"), detail=response["error"])
+
+def say_hi_user(channel_id, user_id):
+  """
+  Sends Hi<user_id> message in a specific channel.
 
+  """
   bot_message = 'Hi <@{}> :wave:'.format(user_id)
-  return slack_client.api_call(api_method='chat.postMessage', json={'channel': channel, 'text': bot_message})
+  return slack_client.api_call(api_method='chat.postMessage', json={'channel': channel_id, 'text': bot_message})
+
+
+def make_gist_payload(url, statement, dialect, created_by):
+  gist_payload = {
+    url: {
+      "color": "#025BA6",
+      "blocks": [
+        {
+          "type": "section",
+          "text": {
+            "type": "mrkdwn",
+            "text": "\n*<{}|Hue - SQL Gist>*".format(url)
+          }
+        },
+        {
+          "type": "section",
+          "text": {
+            "type": "mrkdwn",
+            "text": statement if len(statement) < 150 else (statement[:150] + '...')
+          }
+        },
+        {
+          "type": "section",
+          "fields": [
+            {
+              "type": "mrkdwn",
+              "text": "*Dialect:*\n{}".format(dialect)
+            },
+            {
+              "type": "mrkdwn",
+              "text": "*Created By:*\n{}".format(created_by)
+            }
+          ]
+        }
+      ]
+    }
+  }
+  return gist_payload
+
+
+def make_query_history_payload(url, statement, dialect, database):
+  payload = {
+    url: {
+      "color": "#025BA6",
+      "blocks": [
+        {
+          "type": "section",
+          "text": {
+            "type": "mrkdwn",
+            "text": "\n*<{}|Hue - SQL Editor>*".format(url)
+          }
+        },
+        {
+          "type": "section",
+          "text": {
+            "type": "mrkdwn",
+            "text": statement if len(statement) < 150 else (statement[:150] + '...')
+          }
+        },
+        {
+          "type": "section",
+          "fields": [
+            {
+              "type": "mrkdwn",
+              "text": "*Dialect:*\n{}".format(dialect)
+            },
+            {
+              "type": "mrkdwn",
+              "text": "*Database:*\n{}".format(database)
+            }
+          ]
+        }
+      ]
+    }
+  }
+  return payload
```

Comment:
Same, could even be one 'Query Preview' or small parameterization

Reply:
But won't this bound us for having same type of unflur UI for both gist and query history? @romainr  Also when result will be appended in query history

Reply:
What is the display difference between a gist and query history?
Google 'parameterization' too if really needed, but probably simpler without ;)

Reply:
Aggregated